src/components/auth/implementation/base.ts

```diff
@@ -108,16 +109,23 @@ export async function register(
 export async function session(
   components: Components,
   authedUsername: Maybe<string>,
-  config: Configuration
+  config: Configuration,
+  eventEmitters: { session: Events.Emitter<Events.Session<Session>> }
 ): Promise<Maybe<Session>> {
   if (authedUsername) {
-    return new Session({
+    const session = new Session({
       crypto: components.crypto,
       storage: components.storage,
+      eventEmitter: eventEmitters.session,
       type: TYPE,
       username: authedUsername
     })
 
+    // Emit session create event
+    eventEmitters.session.emit("session:create", { session })
```

Comment:
Why not call this in the Session constructor so that you only have to `emit` the event in one place?

Reply:
Yeah, that's a great idea. Much simpler! 💯 

Updated to move it into the session constructor.